ksqldb-engine/src/main/java/io/confluent/ksql/util/TransientQueryMetadata.java

```diff
@@ -128,15 +137,9 @@ public void setLimitHandler(final LimitHandler limitHandler) {
     rowQueue.setLimitHandler(limitHandler);
   }
 
-  @Override
-  public void stop() {
-    // for transient queries, anytime they are stopped
-    // we should also fully clean them up
-    close();
-  }
-
   @Override
   public void close() {
+    this.isRunning.set(false);
```

Comment:
I think this used to happen at the very end of `super.close()` as part of the close callback and now it's happening at the start. I'm not sure that this is a bug, but might be worth moving it back to the end just in case (this area of the code is very brittle)